op-service/sources/l1_beacon_client.go

```diff
@@ -140,7 +140,7 @@ func (cl *BeaconHTTPClient) BeaconBlobSideCars(ctx context.Context, fetchAllSide
 	}
 
 	if len(indices) > 0 {
-		return eth.APIGetBlobSidecarsResponse{}, fmt.Errorf("#returned blobs(%d) != #requested blobs(%d)", len(hashes)-len(indices), len(hashes))
+		return eth.APIGetBlobSidecarsResponse{}, fmt.Errorf("%s #returned blobs(%d) != #requested blobs(%d)", cl.cl.Endpoint(), len(hashes)-len(indices), len(hashes))
```

Comment:
Ideally we'd log the actual request, not only the endpoint, so what's being built inside `cl.apiReq(ctx, &resp, reqPath, reqQuery)`

Reply:
maybe in apiReq we can access the full request URL from the resp object.

Reply:
Did that here [0f387d3](https://github.com/ethereum-optimism/optimism/pull/14624/commits/0f387d32783868ba7f760c74b1650590322e0539)

I think there is a question about the format of the query params:

`?indices=3&indices=9&indices=11` (current behaviour)

versus

`?indices=3,9,11` (I think seems to match the [spec](https://ethereum.github.io/beacon-APIs/#/Beacon/getBlobSidecars) better, although maybe both are fine). 

I suggest we don't touch the implementation for now and we can then extract queries from the node logs and rerun them against different endpoints ourselves manually.

Reply:
As far as I remember both are valid encodings of the http parameters. So both need to be supported by the blob-api and we should just go with what Go picks by default.